iroh-net/src/discovery/dns.rs

```diff
@@ -36,12 +42,41 @@ const DNS_STAGGERING_MS: &[u64] = &[200, 300];
 #[derive(Debug)]
 pub struct DnsDiscovery {
     origin_domain: String,
+    port: Option<u16>,
+    use_ipv6: bool,
 }
 
 impl DnsDiscovery {
     /// Create a new DNS discovery.
     pub fn new(origin_domain: String) -> Self {
-        Self { origin_domain }
+        Self {
+            origin_domain,
+            port: None,
+            use_ipv6: false,
+        }
+    }
+
+    /// Create a new DNS discovery with a custom port and IP protocol version.
+    ///
+    /// This method configures the DNS query to communicate with the specified DNS server,
+    /// using either IPv4 or IPv6 as determined by the `use_ipv6` flag.
+    ///
+    /// # Arguments
+    ///
+    /// * `origin_domain` - The domain name of the DNS server.
+    /// * `port` - The port number the DNS server is listening on.
+    /// * `use_ipv6` - If true, the query will use IPv6 to connect to the DNS server;
+    ///   if false, it will use IPv4.
+    ///
+    /// Note: To handle both IPv4 and IPv6 connections simultaneously, consider using
+    /// [`ConcurrentDiscovery`](crate::discovery::ConcurrentDiscovery) to combine the queries
+    /// from both versions.
+    pub fn with_port(origin_domain: String, port: u16, use_ipv6: bool) -> Self {
```

Comment:
We have embraced a pattern like this:

fn foo(x) // simple function

fn foo_with_opts(x, opts) // more complex function with lots of settings

Maybe we can do the same here.

fn new(origin_domain) // this would call new_with_opts with the default opts.

fn new_with_opts(origin_domain, opts) // opts contains port, use_ipv6 and possibly more in the future

Reply:
Indeed, this is better.